Dockerfile

```diff
@@ -61,6 +76,17 @@ FROM binaries-$TARGETOS AS binaries
 # enable scanning for this stage
 ARG BUILDKIT_SBOM_SCAN_STAGE=true
 
+FROM gobase AS integration-test-base
+RUN apk add --no-cache docker runc containerd
```

Comment:
As discussed we should provide custom versions in our matrix when running integration tests. e.g., containerd 1.6/1.7 and moby 23/24. Pretty much like we do in BuildKit: https://github.com/moby/buildkit/blob/81d19ad945cdbf921d182c50fc91169b1375b44d/Dockerfile#L4-L6

Can be done in follow-up

Reply:
@crazy-max any chance you could take a look at this one :pray: :pray:

Reply:
Will do this in a follow-up instead, there are more changes than I expected. Some in BuildKit https://github.com/moby/buildkit/compare/master...crazy-max:buildkit:itg-dockerd to set extra envs (basically prepend path with right location like we do with containerd) and path to the alt binary.